galaxy_importer/utils/execution_environment.py

```diff
@@ -0,0 +1,88 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import os
+import requirements
+
+from galaxy_importer.utils import yaml as yaml_utils
+
+
+def process_execution_environment(path, logger):
+    ex_env = {}
+    ex_env_path = os.path.join(path, 'meta', 'execution-environment.yml')
+    ex_env_exists = os.path.exists(ex_env_path)
+
+    if ex_env_exists:
+        ex_env = _load_yaml(ex_env_path, logger)
+
+    if 'dependencies' in ex_env.keys():
+        system_path = os.path.join(path, ex_env['dependencies']['system'])
+        galaxy_path = os.path.join(path, ex_env['dependencies']['galaxy'])
+        python_path = os.path.join(path, ex_env['dependencies']['python'])
+    else:
+        system_path = os.path.join(path, 'bindep.txt')
+        galaxy_path = os.path.join(path, 'requirements.yml')
+        python_path = os.path.join(path, 'requirements.txt')
```

Comment:
It's not a requirement but is the current common location where those files are being included in collections at this time.   I think the intention behind checking the current common location is to make this an easier feature for users to take advantage of.  I'll take it out to avoid any confusion.